### PR TITLE
WRO-2573: Updated Moonstone and samples to React18

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated Moonstone and samples to React18

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-2573

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,16 +39,16 @@
...
-    "@enact/spotlight": "^4.1.2",
-    "@enact/ui": "^4.1.2",
+    "@enact/core": "^4.1.4",
+    "@enact/i18n": "^4.1.4",
+    "@enact/spotlight": "^4.1.4",
+    "@enact/ui": "^4.1.4",
     "classnames": "^2.3.1",
     "invariant": "^2.2.4",
     "prop-types": "^15.8.0",
     "ramda": "^0.27.1",
```
Comment (MikyungKim) [inline package.json]:
```suggestion
    "prop-types": "^15.8.1",
    "ramda": "^0.28.0",
```

File: package.json [hunk 2]
```diff
@@ -39,16 +39,16 @@
...
   },
   "dependencies": {
-    "@enact/core": "^4.1.2",
-    "@enact/i18n": "^4.1.2",
-    "@enact/spotlight": "^4.1.2",
-    "@enact/ui": "^4.1.2",
+    "@enact/core": "^4.1.4",
+    "@enact/i18n": "^4.1.4",
+    "@enact/spotlight": "^4.1.4",
+    "@enact/ui": "^4.1.4",
```
Comment (MikyungKim) [inline package.json:45]: These should be updated to ^4.5.0-alpha.1 once enact released.

File: package.json [hunk 3]
```diff
@@ -39,16 +39,16 @@
...
     "classnames": "^2.3.1",
     "invariant": "^2.2.4",
     "prop-types": "^15.8.0",
     "ramda": "^0.27.1",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "warning": "^4.0.3",
     "ilib": "^14.11.1"
```
Comment (MikyungKim) [inline package.json]:
```suggestion
    "ilib": "^14.14.0"
```

Comment (MikyungKim) [reply package.json]: "peerDependencies": {
    "ilib": "^14.14.0 || ^14.14.0-webos1"
  },
  "devDependencies": {
    "@enact/docs-utils": "^0.3.6",
    "@enact/ui-test-utils": "^1.0.0-alpha.1",
    "eslint-config-enact-proxy": "^1.0.4",
    "ilib": "^14.14.0"
  }
}

Comment (MikyungKim) [reply package.json]: Please update ilib version in samples directories too.

File: package.json (hunk 2 above)
Comment (daniel-stoian-lgp) [reply package.json:45]: yes, i had this in mind :)

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Pull request merged**